app/controllers/flows_controller.rb

```diff
@@ -0,0 +1,147 @@
+class FlowsController < ApplicationController
+  protect_from_forgery except: :show
```

Comment:
You should be able to delete this line if you fetch the page with the authenticity token (aka CSRF token). The jQuery fetch might solve this for you.